This question already has an answer here:Sometimes, under not reproducible circumstances, my WPF application crashes without any message. The application simply close instantly.Where is the best place to implement the global Try/Catch block. At least I have to implement a messagebox with: "Sorry for the inconvenience ..."You can handle the AppDomain.UnhandledException eventEDIT: actually, this event is probably more adequate: Application.DispatcherUnhandledExceptionYou can trap unhandled exceptions at different levels:You should consider what level you need to trap unhandled exceptions at.Deciding between #2 and #3 depends upon whether you\'re using more than one WPF thread.  This is quite an exotic situation and if you\'re unsure whether you are or not, then it\'s most likely that you\'re not.A quick example of code for Application.Dispatcher.UnhandledException:I added this code in App.xaml.csI use the following code in my WPF apps to show a "Sorry for the inconvenience" dialog box whenever an unhandled exception occurs. It shows the exception message, and asks user whether they want to close the app or ignore the exception and continue (the latter case is convenient when a non-fatal exceptions occur and user can still normally continue to use the app).In App.xaml add the Startup event handler:In App.xaml.cs code add Startup event handler function that will register the global application event handler:In addition to the posts above:will not catch exceptions that are thrown from another thread then the main thread. You have to handle those exceptions on its actual Thread. But if you want to Handle them on your global exception handler you can pass it to the main thread:To supplement Thomas\'s answer, the Application class also has the DispatcherUnhandledException event that you can handle.A complete solution is hereit\'s explained very nice with sample code. However, be careful that it does not close the application.Add the line \nApplication.Current.Shutdown(); \nto gracefully close the app.As mentioned above Application.Current.DispatcherUnhandledException will not catch\n  exceptions that are thrown from another thread then the main thread.That actual depend on how the thread was createdOne case that is not handled by Application.Current.DispatcherUnhandledException is System.Windows.Forms.Timer for which Application.ThreadException can be used to handle these\nif you run Forms on other threads than the main thread you will need to set Application.ThreadException from each such thread Best answer is probably https://stackoverflow.com/a/1472562/601990. Here is some code that shows how to use it: App.xaml.cs